src/Sarif.Multitool/Rules/UriMustBeAbsolute.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Json.Pointer;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class UriMustBeAbsolute : SarifValidationSkimmerBase
```

Comment:
>UriMustBeAbsolute [](start = 17, length = 17)

Note that I named this `UriMustBeAbsolute` in contrast to an existing rule `UrisMustBeValid`. My reasoning is that _all_ URIs must be valid, so we say to the user "(All)UrisMustBeValid". Whereas only certain URIs are required to be absolute, so we say to the user "(This)UriMustBeAbsolute".

Thoughts?

Reply:
Accurately describes what it is, sounds good to me.

---
In reply to: [211458124](https://github.com/Microsoft/sarif-sdk/pull/1040#discussion_r211458124) [](ancestors = 211458124)

Reply:
This is the rule that enforces uris that must be absolute, so agree with Chris.

---
In reply to: [211698838](https://github.com/Microsoft/sarif-sdk/pull/1040#discussion_r211698838) [](ancestors = 211698838,211458124)

Reply:
Not to look a gift horse in the mouth, but I was asking your opinion on the (intentional) use of the singular "Uri" in this rule name, while I used the plural "Uris" in the other rule name.

---
In reply to: [211705887](https://github.com/Microsoft/sarif-sdk/pull/1040#discussion_r211705887) [](ancestors = 211705887,211698838,211458124)

Reply:
I like the distinction. Plural is a rule that applies universally. Singular does not.

---
In reply to: [211707455](https://github.com/Microsoft/sarif-sdk/pull/1040#discussion_r211707455) [](ancestors = 211707455,211705887,211698838,211458124)